### DISCUSSION TITLE
Supported models

### DISCUSSION THREAD

**Discussion by Pavellank** [EvanMulawski/FanControl.Razer#3] [General]
He Evan, I`ve tried your plugin on Blade 15 advanced 2021. Used fan control 4.8 net version as stated in readme, but looks like i wont detect any razer sensors. FYI i`v also killed synapse and throttlestop...aaand btw i`ve previosly unlocked my bios for undervolting, don`t know if it has any effect, but in case u need to know that.
Here is the log:
12.07.2023 14:35:27: [Razer] DeviceManager: Found 8 Razer device(s)
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_02#7&31fcfe14&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_00#7&f1ef25a&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col02#7&32f6301c&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=, devicePath=\\?\hid#vid_1532&pid_0276&mi_00&col03#8&1c34417c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col04#7&32f6301c&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col05#7&32f6301c&0&0004#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col03#7&32f6301c&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col01#7&32f6301c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd

12.07.2023 14:35:27: [Razer] DeviceManager: Found 0 supported Razer device(s)

12.07.2023 14:35:30: NvAPIWrapper.Native.Exceptions.NVIDIAApiException: NVAPI_NOT_SUPPORTED
   в NvAPIWrapper.GPU.GPUCoolerInformation.<get_Coolers>d__2.MoveNext()
   в System.Linq.Enumerable.<SelectIterator>d__5`2.MoveNext()
   в System.Linq.Enumerable.<SelectManyIterator>d__19`2.MoveNext()
   в System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   в System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   в FanControl.Domain.BackendProviders.NVApiWrapper.NVApiWrapperBackendProvider.Open()
12.07.2023 14:35:33: [Razer] DeviceManager: Found 8 Razer device(s)
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_02#7&31fcfe14&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_00#7&f1ef25a&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col02#7&32f6301c&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=, devicePath=\\?\hid#vid_1532&pid_0276&mi_00&col03#8&1c34417c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col04#7&32f6301c&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col05#7&32f6301c&0&0004#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col03#7&32f6301c&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col01#7&32f6301c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd

12.07.2023 14:35:34: [Razer] DeviceManager: Found 0 supported Razer device(s)

12.07.2023 14:35:35: NvAPIWrapper.Native.Exceptions.NVIDIAApiException: NVAPI_NOT_SUPPORTED
   в NvAPIWrapper.GPU.GPUCoolerInformation.<get_Coolers>d__2.MoveNext()
   в System.Linq.Enumerable.<SelectIterator>d__5`2.MoveNext()
   в System.Linq.Enumerable.<SelectManyIterator>d__19`2.MoveNext()
   в System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   в System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   в FanControl.Domain.BackendProviders.NVApiWrapper.NVApiWrapperBackendProvider.Open()
12.07.2023 14:35:35: Razer could not initialize or has no sensors.
NVApi could not initialize or has no sensors.
12.07.2023 14:36:15: [Razer] DeviceManager: Found 8 Razer device(s)
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_02#7&31fcfe14&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_00#7&f1ef25a&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col02#7&32f6301c&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=, devicePath=\\?\hid#vid_1532&pid_0276&mi_00&col03#8&1c34417c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col04#7&32f6301c&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col05#7&32f6301c&0&0004#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col03#7&32f6301c&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=Razer Blade, devicePath=\\?\hid#vid_1532&pid_0276&mi_01&col01#7&32f6301c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\kbd

12.07.2023 14:36:15: [Razer] DeviceManager: Found 0 supported Razer device(s)

**Comment by arlo-infinity** [comment]
@Pavellank 

I assume it does, but when you run Razer Synapse can it properly detect the PWM Controller? If no then there are more fundamental issues, but if the answer is yes you may want to try updating the firmware to see if that helps.  I am running device firmware 1.01.00 and it works for me.  To make sure you have the latest firmware search for "Razer PWM PC Fan Controller Firmware Updater" and you should find a link to a Razer page for the download.

**Comment by EvanMulawski** [comment]
Currently, the only supported model is the PWM PC Fan Controller (https://www.razer.com/gaming-pc-accessories/razer-pwm-pc-fan-controller). Note the `DeviceManager: Found 0 supported Razer device(s)` entry in the log file. As your device is a Razer Blade laptop, this plugin will not provide any benefit.
